pysindy/optimizers/sindymodified_optimizer.py

```diff
@@ -0,0 +1,276 @@
+
+    feature_library : feature library object
+        Feature library object used to specify candidate right-hand side features.
+        This must be a class extending
+        :class:`pysindy.feature_library.base.BaseFeatureLibrary`.
+        This is required for modified SINDy because we need to recompute
+        the feature library during each iteration.
+
+    Attributes
+    ----------
+    coef_ : array, shape (n_features,) or (n_targets, n_features)
+        Regularized weight vector(s). This is the v in the objective
+        function.
+
+    unbias : boolean
+        Whether to perform an extra step of unregularized linear regression
+        to unbias the coefficients for the identified support.
+        ``unbias`` is automatically set to False if a constraint is used and
+        is otherwise left uninitialized.
+    """
+
+    def __init__(
+        self,
+        threshold=0.1,
+        tol=1e-5,
+        thresholder="l1",
+        max_iter=10000,
+        normalize=False,
+        fit_intercept=False,
+        copy_X=True,
+        thresholds=None,
+        differentiation_method=None,
+        t=None,
+        optimization_method=tf.keras.optimizers.Adam,
+        learning_rate=0.001,
+        soft_start=False,
+        q=3,
+        c=0.9,
+        feature_library=None,
+    ):
+        super(SINDyModifiedoptimizer, self).__init__(
+            threshold=threshold,
+            tol=tol,
+            thresholder=thresholder,
+            max_iter=max_iter,
+            normalize=normalize,
+            fit_intercept=fit_intercept,
+            copy_X=copy_X,
+        )
+
+        if max_iter <= 0:
+            raise ValueError("max_iter must be positive")
+        if threshold < 0.0:
+            raise ValueError("threshold must not be negative")
+        if thresholder != "l1":
+            raise ValueError("l0 and other nonconvex regularizers are not implemented ")
+        if thresholder[:8].lower() == "weighted" and thresholds is None:
+            raise ValueError("weighted thresholder requires the thresholds parameter")
+        if thresholder[:8].lower() != "weighted" and thresholds is not None:
+            raise ValueError(
+                "The thresholds argument cannot be used without a weighted"
+                " thresholder, e.g. thresholder='weighted_l0'"
+            )
+        if thresholds is not None and np.any(thresholds < 0):
+            raise ValueError("thresholds cannot contain negative entries")
+        if differentiation_method is None:
+            differentiation_method = FiniteDifference(drop_endpoints=False)
+        self.differentiation_method = differentiation_method
+        self.thresholds = thresholds
+        self.reg = get_regularization(thresholder)
+        self.unbias = False
+        self.optimization_method = optimization_method
+        self.learning_rate = learning_rate
+        self.soft_start = soft_start
+        self.q = q
+        self.c = c
+        self.t = t
```

Comment:
Are there more descriptive names we can use for these variables or are you following what's used in the paper?